android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -375,9 +380,10 @@ constructor(
       if (fhirPathExpression.isEmpty()) {
         return emptyList()
       }
-      return resources?.filter {
-        fhirPathDataExtractor.extractValue(it, fhirPathExpression).toBoolean()
-      }
+      return resources
+        ?.asSequence()
```

Comment:
I don't think a sequence will add any performance improvement here there is only one filter operation. Collecting result for long Lists is not performant on a sequence.